cdap-api-common/src/main/java/co/cask/cdap/api/common/Bytes.java

```diff
@@ -896,6 +903,145 @@ int compareTo(T buffer1, int offset1, int length1,
       }
     }
 
+    /**
+     * Copied from guava UnsignedBytes with modification to cater for non-zero byte[] offset and length.
+     */
+    enum UnsafeComparer implements Comparer<byte[]> {
+      INSTANCE;
+
+      static final boolean BIG_ENDIAN = ByteOrder.nativeOrder().equals(ByteOrder.BIG_ENDIAN);
+
+      /*
+       * The following static final fields exist for performance reasons.
+       *
+       * In UnsignedBytesBenchmark, accessing the following objects via static final fields is the
+       * fastest (more than twice as fast as the Java implementation, vs ~1.5x with non-final static
+       * fields, on x86_32) under the Hotspot server compiler. The reason is obviously that the
+       * non-final fields need to be reloaded inside the loop.
+       *
+       * And, no, defining (final or not) local variables out of the loop still isn't as good
+       * because the null check on the theUnsafe object remains inside the loop and
+       * BYTE_ARRAY_BASE_OFFSET doesn't get constant-folded.
+       *
+       * The compiler can treat static final fields as compile-time constants and can constant-fold
+       * them while (final or not) local variables are run time values.
+       */
+
+      static final Unsafe UNSAFE;
+
+      /** The offset to the first element in a byte array. */
+      static final int BYTE_ARRAY_BASE_OFFSET;
+
+      private static final int UNSIGNED_MASK = 0xFF;
+
+      static {
+        UNSAFE = getUnsafe();
+
+        BYTE_ARRAY_BASE_OFFSET = UNSAFE.arrayBaseOffset(byte[].class);
+
+        // sanity check - this should never fail
+        if (UNSAFE.arrayIndexScale(byte[].class) != 1) {
+          throw new AssertionError();
+        }
+      }
+
+      /**
+       * Returns a sun.misc.Unsafe. Suitable for use in a 3rd party package. Replace with a simple
+       * call to Unsafe.getUnsafe when integrating into a jdk.
+       *
+       * @return a sun.misc.Unsafe
+       */
+      private static sun.misc.Unsafe getUnsafe() {
```

Comment:
Because the Unsafe class is a internal class which doesn't exist on all JVM implementation. Having an import statement has no impact on the compiled code.